components/service/glean/docs/metrics/adding-new-metrics.md

```diff
@@ -108,3 +108,50 @@ assertEquals(6, Controls.refreshPressed.testGetValue())
 
 NOTE: Using the testing API requires calling `Glean.enableTestingMode()` first,
 such as from a `@Before` method.
+
+## Labeled metrics
+
+Some metrics can be used as *labeled* variants. This means that for a single metric entry you define in `metrics.yaml`, you can record into multiple metrics under the same name, each identified by a different string label.
+
+This is useful when you need to break down metrics by a label known at build time or run time. For example:
+- When you want to count a different set of subviews that users interact with, you could use `viewCount["view1"].add()` and `viewCount["view2"].add()`.
+- When you want to count errors that might occur for a feature, you could use `errorCount[errorName].add()`.
+
+**Note**: Be careful with using arbitrary strings as labels and make sure they can't accidentally contain identifying data (like directory paths or user input).
+
+Say you're adding a new counter for errors that can occur when loading a resource from a REST API. First you need to add an entry for the counter to the `metrics.yaml` file:
+
+```YAML
+updater:
+  load_error:
+    type: counter
+    labeled: true # This makes it a labeled counter.
+    labels: # This is optional, if provided it limits the set of labels you can use.
+    - timeout
+    - not_found
+    description: >
+      Counts the different types of load errors that can occur.
+    ...
+```
+
+Now you can use the labeled counter from the applications code:
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Updater
+
+// ...
+Updater.loadError["timeout"].add() // Adds 1 to the "timeout" counter.
+Updater.loadError["not_found"].add(2)
+```
```

Comment:
Should we add:

"If the labels are predefined in the metrics.yaml and a label not in that set it used, the value is applied to a special `__other__` category."

Reply:
Hm, i would prefer to keep it out of this document to keep things simple.
This would be important for the longer article for the metric.

Reply:
Ok.